eppo_core/src/events/event_delivery.rs

```diff
@@ -0,0 +1,63 @@
+use crate::events::event::Event;
+use crate::{Error, Str};
+use log::{debug, info};
+use reqwest::{StatusCode};
+use serde::{Deserialize, Serialize};
+use url::{ParseError, Url};
+use uuid::Uuid;
+
+#[derive(Clone)]
+pub struct EventDelivery {
+    sdk_key: Str,
+    ingestion_url: Url,
+    client: reqwest::Client,
+}
+
+#[derive(serde::Deserialize)]
+pub struct EventDeliveryResponse {
+    pub failed_events: Vec<Uuid>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct IngestionRequestBody {
+    eppo_events: Vec<Event>,
+}
+
+/// Responsible for delivering event batches to the Eppo ingestion service.
+impl EventDelivery {
+    pub fn new(sdk_key: String, ingestion_url: String) -> Result<Self, ParseError> {
```

Comment:
minor: in rust when taking ownership, it's preferable to ask for exact types that you need (e.g., `Str` and `Url` here). If the caller has the appropriate types, it can pass the values in as is (instead of potentially converting `Str` to `String`, only to get converted back to `Str`)
```suggestion
    pub fn new(sdk_key: Str, ingestion_url: Url) -> Self {
```

Reply:
I added it after you merged, so no worries :)